nixos/modules/services/networking/wpa_supplicant.nix

```diff
@@ -107,6 +107,10 @@ let
       stopIfChanged = false;
 
       path = [ package ];
+      # if `userControl.enable`, the supplicant automatically changes the permissions
+      #  and owning group of the runtime dir; setting `umask` ensures the generated
+      #  config file isn't readable (except to root);  see nixpkgs#267693
+      serviceConfig.UMask = "066";
```

Comment:
Why use "6" and allow potentially creating files executable by the world? Isn't it best to use "7"? (I don't care about the group perms, but world/other perms.)

Reply:
The first thing that comes to mind is that this also affects `mkdir` and we might want to create directories whose contents non-root users can access. But given that they won't be able to list them due to the read bit being masked out, I don't think it's super relevant.

At the same time, this prevents creation of executable filesystem nodes, but the executable mode would have to be specified explicitly anyway in the call to whichever syscall of the `open` family is used. I don't think it's likely that this would happen by accident, and don't think it _needs_ to be changed. But if you care a lot and want to open a PR you're welcome to :)

Reply:
Ok, thanks for the input.